airflow/providers/snowflake/transfers/copy_into_snowflake.py

```diff
@@ -0,0 +1,146 @@
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""This module contains abstract operator that child classes implements
+COPY INTO <TABLE> SQL in Snowflake
+"""
+
+
+from typing import List, Optional, Sequence
+
+from airflow.models import BaseOperator
+from airflow.providers.snowflake.hooks.snowflake import SnowflakeHook
+from airflow.providers.snowflake.utils.common import enclose_param
+from airflow.utils.context import Context
+
+
+class CopyFromExternalStageToSnowflakeOperator(BaseOperator):
+    """
+    Executes a COPY INTO command to load files from an external stage from clouds to Snowflake
+
+    This operator requires the snowflake_conn_id connection. The snowflake host, login,
+    and, password field must be setup in the connection. Other inputs can be defined
+    in the connection or hook instantiation.
+
+    :param namespace: snowflake namespace
+    :param table: snowflake table
+    :param file_format: file format name i.e. CSV, AVRO, etc
+    :param stage: reference to a specific snowflake stage. If the stage's schema is not the same as the
+        table one, it must be specified
+    :param prefix: cloud storage location specified to limit the set of files to load
+    :param files: files to load into table
+    :param pattern: pattern to load files from external location to table
+    :param copy_into_postifx: optional sql postfix for INSERT INTO query
+           such as `formatTypeOptions` and `copyOptions`
+    :param snowflake_conn_id:  Reference to :ref:`Snowflake connection id<howto/connection:snowflake>`
+    :param account: snowflake account name
+    :param warehouse: name of snowflake warehouse
+    :param database: name of snowflake database
+    :param region: name of snowflake region
+    :param role: name of snowflake role
+    :param schema: name of snowflake schema
+    :param authenticator: authenticator for Snowflake.
+        'snowflake' (default) to use the internal Snowflake authenticator
+        'externalbrowser' to authenticate using your web browser and
+        Okta, ADFS or any other SAML 2.0-compliant identify provider
+        (IdP) that has been defined for your account
+        ``https://<your_okta_account_name>.okta.com`` to authenticate
+        through native Okta.
+    :param session_parameters: You can set session-level parameters at
+        the time you connect to Snowflake
+
+    """
+
+    template_fields: Sequence[str] = ("files",)
+    template_fields_renderers = {"files": "json"}
+
+    def __init__(
+        self,
+        table: str,
+        file_format: str,
+        stage: str,
+        stage_postfix: Optional[str] = None,
+        columns_array: Optional[List[str]] = None,
+        files: Optional[List[str]] = None,
+        pattern: Optional[str] = None,
+        copy_into_postfix: Optional[str] = None,
```

Comment:
Might be better use `copy_options` and `validation_mode` in order to conform with [Snowflake COPY INTO syntax](https://docs.snowflake.com/en/sql-reference/sql/copy-into-table.html#syntax)